LIS/LISAv2
# azure,hyperv: Added 10 functional SR-IOV test cases

This adds 9 new TCs for both Azure and Hyper-V:
--VERIFY-SRIOV-LSPCI
--SRIOV-VERIFY-VF-BASIC-CONNECTION
--SRIOV-VERIFY-VF-BASIC-CONNECTION-MAX-VCPU
--SRIOV-VERIFY-VF-MULTIPLE-CONNECTION
--SRIOV-VERIFY-VF-MAX-CONNECTION
--SRIOV-VERIFY-VF-MAX-CONNECTION-MAX-VCPU
--SRIOV-IPERF-STRESS
--SRIOV-DISABLEVF-ON-GUEST
--SRIOV-RELOAD-MODULE

One Hyper-V only TC was added also:
--SRIOV-VERIFY-VF-BASIC-CONNECTION-ONE-VCPU

One existing TC was modified and was added HyperV support:
--VERIFY-SRIOV-LSPCI -> sriov_lspci.sh was removed and
SRIOV-Verify-lspci.sh will now be the test script for this test case

One existing TC was removed because more comprehensive Azure/HyperV test
cases were added:
--VERIFY-IFUP-IFDOW-ETH0-RMMOD-MODPROBE-HV_NETVSC-SRIOV use case is now
done by SRIOV-DISABLEVF-ON-GUEST and SRIOV-RELOAD-MODULE test cases

### PS Script Analyzer was run

# Test results

## Azure
ARM Image under test **RedHat : RHEL : 7-RAW : latest**
Total Executed TestCases          9 (         7 Pass,          0 Fail,          2 Abort)
Total Execution Time(dd:hh:mm) 0:2:35
XML file: TestConfiguration

Sr. Test Name : Test Result : Test Duration (in minutes)
----------------------------------------------------
1. VERIFY-SRIOV-LSPCI : PASS : 3.01
2. SRIOV-VERIFY-VF-BASIC-CONNECTION : PASS : 4.09
3. SRIOV-VERIFY-VF-BASIC-CONNECTION-MAX-VCPU : PASS : 4.06
4. SRIOV-VERIFY-VF-MULTIPLE-CONNECTION : PASS : 4.6
5. SRIOV-VERIFY-VF-MAX-CONNECTION : ABORTED : 42.63
6. SRIOV-VERIFY-VF-MAX-CONNECTION-MAX-VCPU : ABORTED : 42.93
7. SRIOV-IPERF-STRESS : PASS : 39.35
8. SRIOV-DISABLEVF-ON-GUEST : PASS : 4.68
9. SRIOV-RELOAD-MODULE : PASS : 4.75

ARM Image under test **Canonical : UbuntuServer : 18.04-LTS : latest**
Total Executed TestCases            9 (           7 Pass,            0 Fail,            2 Abort)
Total Execution Time(dd:hh:mm) 0:2:58
XML file: TestConfiguration

Sr. Test Name : Test Result : Test Duration (in minutes)
----------------------------------------------------
1. VERIFY-SRIOV-LSPCI : PASS : 3.6
2. SRIOV-VERIFY-VF-BASIC-CONNECTION : PASS : 4.37
3. SRIOV-VERIFY-VF-BASIC-CONNECTION-MAX-VCPU : PASS : 5.45
4. SRIOV-VERIFY-VF-MULTIPLE-CONNECTION : PASS : 14.74
5. SRIOV-VERIFY-VF-MAX-CONNECTION : ABORTED : 41.09
6. SRIOV-VERIFY-VF-MAX-CONNECTION-MAX-VCPU : ABORTED : 62.67
7. SRIOV-IPERF-STRESS : PASS : 36.38
8. SRIOV-DISABLEVF-ON-GUEST : PASS : 5.18
9. SRIOV-RELOAD-MODULE : PASS : 4.97

**SRIOV-VERIFY-VF-MAX-CONNECTION and SRIOV-VERIFY-VF-MAX-CONNECTION-MAX-VCPU are failing because of existing issues with VM sizes with 8 NICs**

## Hyper-V
VHD under test net_vmu640master.vhdx
ARM Image under test  :  :  :
Total Executed TestCases        10 (       10 Pass,        0 Fail,        0 Abort)
Total Execution Time(dd:hh:mm) 0:1:27
XML file: TestConfiguration

Sr. Test Name : Test Result : Test Duration (in minutes)
----------------------------------------------------
1. VERIFY-SRIOV-LSPCI : PASS : 3.63
2. SRIOV-VERIFY-VF-BASIC-CONNECTION : PASS : 6.1
3. SRIOV-VERIFY-VF-BASIC-CONNECTION-ONE-VCPU : PASS : 4.39
4. SRIOV-VERIFY-VF-BASIC-CONNECTION-MAX-VCPU : PASS : 5.25
5. SRIOV-VERIFY-VF-MULTIPLE-CONNECTION : PASS : 6.87
6. SRIOV-VERIFY-VF-MAX-CONNECTION : PASS : 8.1
7. SRIOV-VERIFY-VF-MAX-CONNECTION-MAX-VCPU : PASS : 7.3
8. SRIOV-IPERF-STRESS : PASS : 36.27
9. SRIOV-DISABLEVF-ON-GUEST : PASS : 4.63
10. SRIOV-RELOAD-MODULE : PASS : 5.28

VHD under test ubuntu_18.04_x64.vhd
ARM Image under test  :  :  :
Total Executed TestCases             10 (            8 Pass,             0 Fail,             2 Abort)
Total Execution Time(dd:hh:mm) 0:1:58
XML file: TestConfiguration

Sr. Test Name : Test Result : Test Duration (in minutes)
----------------------------------------------------
1. VERIFY-SRIOV-LSPCI : PASS : 5.84
2. SRIOV-VERIFY-VF-BASIC-CONNECTION : PASS : 8.85
3. SRIOV-VERIFY-VF-BASIC-CONNECTION-ONE-VCPU : PASS : 5.67
4. SRIOV-VERIFY-VF-BASIC-CONNECTION-MAX-VCPU : ABORTED : 7.26
5. SRIOV-VERIFY-VF-MULTIPLE-CONNECTION : PASS : 11.88
6. SRIOV-VERIFY-VF-MAX-CONNECTION : PASS : 19.13
7. SRIOV-VERIFY-VF-MAX-CONNECTION-MAX-VCPU : ABORTED : 7.46
8. SRIOV-IPERF-STRESS : PASS : 38.68
9. SRIOV-DISABLEVF-ON-GUEST : PASS : 6.01
10. SRIOV-RELOAD-MODULE : PASS : 8

**SRIOV-VERIFY-VF-BASIC-CONNECTION-MAX-VCPU and SRIOV-VERIFY-VF-MAX-CONNECTION-MAX-VCPU are failing on Bionic because of an existing issue with max vCPU (increased boot time)**
